consensus/src/core.rs

```diff
@@ -0,0 +1,217 @@
+use crate::config::{Committee, Parameters};
+use crate::context::Context;
+use crate::error::ConsensusResult;
+use crate::timer::Timer;
+use bft_lib::base_types::NodeTime;
+use bft_lib::interfaces::{ConsensusNode, DataSyncNode, NodeUpdateActions};
+use bft_lib::smr_context::SmrContext;
+use bytes::Bytes;
+use crypto::{PublicKey, SignatureService};
+use futures::executor::block_on;
+use librabft_v2::data_sync::{DataSyncNotification, DataSyncRequest, DataSyncResponse};
+use log::{debug, warn};
+use network::NetMessage;
+use serde::{de::DeserializeOwned, Deserialize, Serialize};
+use std::fmt::Debug;
+use std::time::{SystemTime, UNIX_EPOCH};
+use store::Store;
+use tokio::sync::mpsc::{Receiver, Sender};
+
+// TODO: Temporarily disable tests.
+// #[cfg(test)]
+// #[path = "tests/core_tests.rs"]
+// pub mod core_tests;
+
+pub type RoundNumber = u64;
+
+#[derive(Serialize, Deserialize, Debug)]
+pub enum ConsensusMessage {
+    DataSyncNotification {
+        sender: PublicKey,
+        notification: DataSyncNotification<Context>,
```

Comment:
So currently, the (non-simulated) Context only works for LibraBFTv2 messages?

Reply:
No that's bad, I have to fix it